modules/grids/app/components/grids/widgets/subitems.rb

```diff
@@ -61,15 +61,23 @@ def wrapper_arguments
         { full_width: true }
       end
 
+      def can_view_subprojects?
+        current_user.allowed_in_project?(:view_project, project)
```

Comment:
This checks wether the user is allowed to see the current project not any of its subprojects.

Reply:
Actually, we check this permission [here](https://github.com/opf/openproject/blob/dev/modules/grids/lib/grids/engine.rb#L15)
Also [here](https://github.com/opf/openproject/blob/dev/modules/grids/app/components/grids/widgets/subitems.rb#L66), we return the subitems that the current user has access to.